tests/fracs/test_fracture_network_3d.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+import pytest
+
+import porepy as pp
+from porepy.applications.test_utils.arrays import compare_arrays
+from porepy.fracs.plane_fracture import PlaneFracture
+from porepy.geometry.domain import Domain
+from porepy.grids.standard_grids.utils import unit_domain
+
+
+# ---------- Testing impose_external_boundary ----------
+# Test of algorithm for constraining a fracture a bounding box.
+
+# Since that algorithm uses fracture intersection methods, the tests functions as
+# partial test for the wider fracture intersection framework as well. Full tests
+# of the latter are too time consuming to fit into a unit test.
+
+# Now the boundary is defined as set of "fake" fractures, all fracture network
+# have 2*dim additional fractures (hence the + 6 in the assertions)
+
+
+@pytest.fixture
+def domain() -> Domain:
+    return unit_domain(3)
+
+
+@pytest.fixture
+def fracture() -> PlaneFracture:
+    return PlaneFracture(
+        np.array([[0, 1, 1, 0], [0.5, 0.5, 0.5, 0.5], [0, 0, 1, 1]]),
+        check_convexity=False,
+    )
+
+
+def test_completely_outside_lower(domain: Domain, fracture: PlaneFracture):
```

Comment:
We can parametrize most of these functions, can we not?

Reply:
Agreed, changing it